### PR TITLE
non ansi join implemented

### PR DESCRIPTION
**Pull request by sriram251-code**
fixes #846

### CODE REVIEW

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4 [hunk 1]
```diff
@@ -3126,7 +3126,8 @@ exprList: expr (COMMA expr)*
     ;
 
 expr
-    : L_PAREN expr R_PAREN                      # exprPrecedence
+    :
+    L_PAREN expr R_PAREN                      # exprPrecedence
```
Comment (nfx) [inline core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4:3130]: why this line was changed?

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4 [hunk 2]
```diff
@@ -3149,6 +3150,7 @@ expr
     | DISTINCT expr                             # exprDistinct
     | L_PAREN subquery R_PAREN                  # exprSubquery
     | primitiveExpression                       # exprPrimitive
+    | expr L_PAREN PLUS R_PAREN                 #exprJoin
```
Comment (nfx) [inline core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4:3153]:
```suggestion
    | expr L_PAREN PLUS R_PAREN                 # exprJoin
```

probably this needs to be moved up

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala [hunk 3]
```diff
@@ -572,4 +572,9 @@ class SnowflakeExpressionBuilder()
   override def visitExprSubquery(ctx: ExprSubqueryContext): ir.Expression = {
     ir.ScalarSubquery(ctx.subquery().accept(new SnowflakeRelationBuilder))
   }
+
+  override def visitExprJoin(ctx: ExprJoinContext): ir.Expression = {
+    ir.JoinMarkExpression(ctx.expr().accept(this))
+
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala:578]: remove blank line

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala [hunk 4]
```diff
@@ -238,3 +238,7 @@ case class Variable(name: String) extends LeafExpression {
 case class SchemaReference(columnName: Expression) extends Unary(columnName) {
   override def dataType: DataType = UnresolvedType
 }
+
+case class JoinMarkExpression(value: Expression) extends LeafExpression {
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala:242]: how is it named in catalyst?

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeRelationBuilder.scala [hunk 5]
```diff
@@ -71,12 +72,15 @@ class SnowflakeRelationBuilder
     }
 
   override def visitSelectOptionalClauses(ctx: SelectOptionalClausesContext): ir.LogicalPlan = {
+
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeRelationBuilder.scala:75]: revert

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala [hunk 6]
```diff
@@ -6,6 +6,19 @@ class SnowflakeToDatabricksTranspilerTest extends AnyWordSpec with TranspilerTes
 
   protected val transpiler = new SnowflakeToDatabricksTranspiler
 
+  "Snowflake non ansi command" should {
+    "transpile non ansi command" in {
+      "select * from t1,t2 where t1.x=t2.x(+) " transpilesTo
```
Comment (nfx) [inline core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala:11]:
```suggestion
    "select * from t1,t2 where t1.x=t2.x(+)" in {
      "select * from t1,t2 where t1.x=t2.x(+)" transpilesTo
```
follow the style of other tests

**Review by nfx — CHANGES_REQUESTED** [review]

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4 (hunk 1 above)
Comment (sriram251-code) [reply core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4:3130]: will take care of it. Its work in progress and a draft PR, As a practice I commit the dev code on regular basis.

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4 (hunk 2 above)
Comment (sriram251-code) [reply core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4:3153]: aggreed

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala (hunk 3 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala:578]: done, will take care of it. Its work in progress and a draft PR, As a practice I commit the dev code on regular basis.

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala (hunk 4 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala:242]: Its work in progress and a draft PR, As a practice I commit the dev code on regular basis.

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeRelationBuilder.scala (hunk 5 above)
Comment (sriram251-code) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeRelationBuilder.scala:75]: Done. Its work in progress and a draft PR, As a practice I commit the dev code on regular basis.

File: core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala (hunk 6 above)
Comment (sriram251-code) [reply core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala:11]: will take care of it. Its work in progress and a draft PR, As a practice I commit the dev code on regular basis.

**Comment by sriram251-code** [comment]
The implementation of sqlQlot has been completed.
 However, before proceeding with the implementation of non-ANSI joins in Antlr, an important prerequisite needs to be addressed. Specifically, we need to develop a feature that can resolve references within a query.

**Reference Resolution**
This feature is crucial for resolving references in complex queries. For example, detecting in a query like:
`WHERE schema.t1.employee = schema.t2.employee AND schema.t1.department = schema.t3.department` involves t1,t2 and t3.

**Challenges with Expression Resolution**
Challenges with Expression Resolution
Another challenge that has been identified is the resolution of expressions without aliases. For instance, in a condition like:
`SAL(+) = 100`
Proper resolution of this expression requires access to the schema of the relevant tables. Currently, this schema information is not available in the system.

Conclusion
Due to these unresolved prerequisites, particularly the need for reference resolution and schema availability, the implementation of non-ANSI joins cannot proceed at this time. As a result, the current pull request (PR) is being closed.
@vil1 @nfx @jimidle @gueniai

**Pull request closed without merging**